builder/pipeline.cc

```diff
@@ -250,6 +250,23 @@ class sanitize_user_exprs : public node_mutator {
     return let_stmt::make(std::move(lets), std::move(s));
   }
 
+  void visit(const variable* op) override {
+    if (op->field == buffer_field::none) {
+      node_mutator::visit(op);
+      return;
+    }
+
+    // Don't lift internally allocated buffer metadata expressions.
+    // TODO: This should be a proper API error.
+    assert(std::binary_search(external.begin(), external.end(), op->sym));
```

Comment:
If this should be a real API error, should we check in all runtime modes?

Reply:
Yes but this is just duplicating a pre-existing check like this (from `call` visitor). Let's leave it for now.